cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
FYI: This is really the only comment I care about resolving before merging

From http://kubernetes.io/docs/user-guide/config-best-practices/:

> Don’t use hostPort (which specifies the port number to expose on the host) unless absolutely necessary, e.g., for a node daemon. When you bind a Pod to a hostPort, there are a limited number of places that pod can be scheduled, due to port conflicts— you can only schedule as many such Pods as there are nodes in your Kubernetes cluster. If you only need access to the port for debugging purposes, you can use the kubectl proxy and apiserver proxy or kubectl port-forward. You can use a Service object for external service access. If you do need to expose a pod’s port on the host machine, consider using a NodePort service before resorting to hostPort. Avoid using hostNetwork, for the same reasons as hostPort.

So the issue is:

> I tracked it down to the attempt to inherit DNS from the host: this allows us to use the site's DNS to resolve names that are not maintained by SkyDNS.

This is the correct behavior of skydns and should not be slow even when using pod network. @ArtfulCoder, @kubernetes/goog-cluster can you advise on how to debug this slowness?